Source/WebKit/Shared/IPCStreamTesterProxy.h

```diff
@@ -44,6 +44,11 @@ class IPCStreamTesterProxy final : public IPC::MessageReceiver {
 public:
     // IPC::MessageReceiver overrides.
     void didReceiveMessage(IPC::Connection&, IPC::Decoder&);
+
+    // Do nothing since this class is never instantiated.
+    void ref() const { }
+    void deref() const { }
```

Comment:
This seems like a risky pattern. If a derived class doesn't implement `ref` and `deref` the code will just silently do the wrong thing?

Reply:
The constructor and destructor are private. There is a comment at the top of the file that this class is never instantiated. I agree it is a bit fragile but it is also testing code. I could make the class RefCounted for good measure but then I'd have to make the destructor public.

Reply:
All good points.